### DISCUSSION TITLE
How can I loop the program for specified period of time

### DISCUSSION THREAD

**Discussion by iamraghav1503** [flypythoncom/python#40] [Q&A]
pe.moveTo(132,475,1),pe.click(132,475,1),pe.typewrite(superb bro i want this because of camera),pe.click(917,515,1)

i want to loop this program for 2 hours and after the word 'camera' in the sentence i want to add non repeating unique numbers

**Comment by ferid333** [comment]
Hello @iamraghav1503 !
To loop this program for 2 hours and append non-repeating unique numbers to the word "camera" in the sentence, you can use Python. Here's an example script that achieves this:
```python
import pyautogui as pe
import time

# Start time
start_time = time.time()

# Initialize a counter for unique numbers
counter = 1

# Run the loop for 2 hours
while time.time() - start_time < 2 * 60 * 60:
    # Perform the actions
    pe.moveTo(132, 475, 1)
    pe.click(132, 475, 1)
    
    # Create the unique sentence
    message = f"superb bro i want this because of camera{counter}"
    pe.typewrite(message)
    
    pe.click(917, 515, 1)
    
    # Increment the counter for unique numbers
    counter += 1
    
    # Optional: Add a delay between iterations if needed (e.g., 1 second)
    time.sleep(1)
```

**Comment by Riya-877** [comment]
Hey! 👋

You can do this using a **time-based loop** in Python along with a counter for unique numbers.

Here’s a simple and clean version:

```python
import pyautogui as pe
import time

start_time = time.time()
counter = 1

# Run for 2 hours (2 * 60 * 60 seconds)
while time.time() - start_time < 7200:
    
    pe.moveTo(132, 475, 1)
    pe.click(132, 475, 1)
    
    # Add unique number after 'camera'
    message = f"superb bro i want this because of camera{counter}"
    pe.typewrite(message)
    
    pe.click(917, 515, 1)
    
    counter += 1
    time.sleep(1)  # small delay to avoid too fast execution

print("Finished running for 2 hours ✅")
```
🔹 Explanation (simple)

* `time.time()` → tracks current time
* Loop runs until 2 hours are completed
* `counter` → ensures unique numbers (camera1, camera2, camera3...)

---
💡 Tip

If you want *random* unique numbers instead of sequential ones, you can use:

```python
import random
random.randint(1000, 9999)
```

---

Hope this helps! Let me know if you want to modify timing or speed 🙂
